app/controllers/lessons_controller.rb

```diff
@@ -24,20 +24,43 @@ def new
     @lesson = Lesson.new
   end
 
+  def edit
+    @lesson = Lesson.find params.require :id
+    authorize @lesson
+  end
+
   def create
     @lesson = Lesson.new params.require(:lesson).permit :group_id, :date, :subject_id
-    authorize @lesson
-    if @lesson.save
+    if @lesson.valid? && @lesson.save
+      authorize @lesson
```

Comment:
shouldn't the authorization happen before save? I'm pretty sure it should. Authorization checks if the current user is allowed to make changes to the lesson and if it's done after the lesson is already saved then it's too late.